components/FormField.tsx

```diff
@@ -20,27 +21,54 @@ export function FormField<T extends FieldValues>({
   label,
   name,
   inputProps,
+  render,
 }: {
   form: UseFormReturn<T>;
   label: ReactNode;
   name: Path<T>;
-  inputProps: InputProps;
+  inputProps?: InputProps;
+  render?: (
+    props: {
+      inputProps: {
+        id: string;
+        onPress: (e: any) => void;
+        opacity: number;
+      };
+    } & Parameters<ControllerProps["render"]>[0]
+  ) => JSX.Element;
 }) {
   return (
     <YStack alignItems="stretch">
       <Label htmlFor={name}>{label}</Label>
       <Controller
         control={form.control}
         name={name}
-        render={({ field: { onChange, ...field } }) => (
-          <Input
-            id={name}
-            {...inputProps}
-            {...field}
-            onChangeText={onChange}
-            opacity={field.disabled ? 0.5 : 1}
-          />
-        )}
+        render={({ field: { onChange, ...field }, ...renderProps }) =>
+          render ? (
+            render({
+              inputProps: {
+                id: name,
+                onPress: (e) => {
+                  e.stopPropagation();
+                },
+                opacity: field.disabled ? 0.5 : 1,
+              },
+              field: { ...field, onChange },
+              ...renderProps,
```

Comment:
So this really should be refactored to be a little simpler, but this works for now.